server/core/src/main/java/org/infinispan/server/core/AbstractCacheIgnoreAware.java

```diff
@@ -11,24 +14,34 @@
  * @since 9.0
  */
 public class AbstractCacheIgnoreAware implements CacheIgnoreAware {
+   private static final String IGNORED_CACHES_KEY = "ignored-caches";
 
-   private Set<String> ignoredCaches = ConcurrentHashMap.newKeySet();
+   private Cache<String, Set<String>> stateCache;
 
-   public void setIgnoredCaches(Set<String> cacheNames) {
-      ignoredCaches.clear();
-      cacheNames.forEach(ignoredCaches::add);
+   public void initialize(Cache<String, Set<String>> stateCache) {
+      this.stateCache = stateCache;
    }
 
    public void unignore(String cacheName) {
-      ignoredCaches.remove(cacheName);
+      Set<String> currentValue = stateCache.get(IGNORED_CACHES_KEY);
+      if (cacheName == null || currentValue == null) return;
+      if (currentValue.remove(cacheName)) stateCache.put(IGNORED_CACHES_KEY, currentValue);
    }
 
    public void ignoreCache(String cacheName) {
-      ignoredCaches.add(cacheName);
+      Set<String> currentValue = stateCache.get(IGNORED_CACHES_KEY);
+      Set<String> newValue = currentValue == null ? new HashSet<>() : currentValue;
+      if (newValue.add(cacheName)) stateCache.put(IGNORED_CACHES_KEY, newValue);
```

Comment:
it doesn't look safe to me.
If you have multiple concurrent ignore requests, you will lose some of them.

Reply:
ps. also, you can use the `Cache` keys as a "ignore" `Set`. It will avoid replicating the full set every time.

Reply:
> ps. also, you can use the Cache keys as a "ignore" Set. 
I need to have it under a single key, since the state cache in theory could be used to store other things that the server may need at runtime

Reply:
> it doesn't look safe to me.

Agreed, this looks bad, will fix it